common/src/main/scala/uk/ac/wellcome/platform/archive/common/bagit/models/ExternalIdentifier.scala

```diff
@@ -3,19 +3,39 @@ package uk.ac.wellcome.platform.archive.common.bagit.models
 import org.scanamo.DynamoFormat
 import io.circe.{Decoder, Encoder, HCursor, Json}
 
-case class ExternalIdentifier(underlying: String) extends AnyVal {
+class ExternalIdentifier(val underlying: String) {
   override def toString: String = underlying
+
+  require(!underlying.isEmpty, "External identifier cannot be empty")
+
+  // Normally we use case classes for immutable data, which provide these
+  // methods for us.
+  //
+  // We deliberately don't use case classes here so we skip automatic
+  // case class derivation for JSON encoding/DynamoDB in Scanamo,
+  // and force callers to intentionally import the implicits below.
```

Comment:
Previously we got this through AnyVal, which we've had to abandon because AnyVal doesn't let you do runtime checks.